easydigitaldownloads/easy-digital-downloads
# Update customer stats via ajax when recalculating single customer

Fixes #8004

Proposed Changes:
1. Adds a new class `$data` variable to `EDD_Tools_Recount_Single_Customer_Stats`.
2. On success, the new purchase count and value are sent to the JavaScript.
3. The `span` which match the data keys are updated with the new values.

My thought on how the script is updating the matching `span` is that this is something that other export classes could use if needed--the array of data returned would just have to set the keys to match an element with a matching class (prepended with `edd_`). In that case, it might be better to define the new `$data` variable in `EDD_Batch_Export`, but this is a start.

To test:
1. Visit the Customer Details screen for any customer.
2. Use your browser's dev tools to change the numbers for the "Completed Sales" and "Lifetime Value". (You could also do this directly in the database, but this felt easier.)
3. Go to Tools for the customer.
4. Click the Recount Stats button.
5. When the success message appears, click on the customer Profile link.
6. The stats should be updated to the new/correct values.
